### PR TITLE
Clear error message when switching tabs

### PR DESCRIPTION
**Pull request by ioedeveloper**
Similar changes will also be needed for https://github.com/ethereum/remix-project/issues/2097

### CODE REVIEW

File: apps/remix-ide/src/app/tabs/compile-tab.js [hunk 1]
```diff
@@ -82,6 +86,7 @@ class CompileTab extends CompilerApiMixin(ViewPlugin) { // implements ICompilerA
   }
 
   getFileManagerMode () {
+    this.emit()
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compile-tab.js:89]: what are you emitting?

File: libs/remix-ui/solidity-compiler/src/lib/solidity-compiler.tsx [hunk 2]
```diff
@@ -40,6 +42,14 @@ export const SolidityCompiler = (props: SolidityCompilerProps) => {
     })()
   }, [])
 
+  useEffect(() => {
+    if (badgeStatus[currentFile]) {
+      api.emit('statusChanged', badgeStatus[currentFile])
```
Comment (yann300) [inline libs/remix-ui/solidity-compiler/src/lib/solidity-compiler.tsx:47]: can't you use the `statusChanged` function too here

File: apps/remix-ide/src/app/tabs/compile-tab.js (hunk 1 above)
Comment (ioedeveloper) [reply apps/remix-ide/src/app/tabs/compile-tab.js:89]: This was a mistake. I will take it out.

File: libs/remix-ui/solidity-compiler/src/lib/solidity-compiler.tsx (hunk 2 above)
Comment (ioedeveloper) [reply libs/remix-ui/solidity-compiler/src/lib/solidity-compiler.tsx:46]: `statusChanged` function wouldn't work here. A plugin emitted event is needed.

**Review by yann300 — APPROVED** [review]

**Pull request merged**